### PR TITLE
Fix possible error if an incorrect version list array is provided

### PR DESCRIPTION
**Pull request by sgiehl**
### Description:

refs https://github.com/matomo-org/matomo/issues/19806

### Review

* [ ] [Functional review done](https://developer.matomo.org/guides/pull-request-reviews#functional-review-done)
* [ ] [Potential edge cases thought about](https://developer.matomo.org/guides/pull-request-reviews#potential-edge-cases-thought-about) (behavior of the code with strange input, with strange internal state or possible interactions with other Matomo subsystems)
* [ ] [Usability review done](https://developer.matomo.org/guides/pull-request-reviews#usability-review-done) (is anything maybe unclear or think about anything that would cause people to reach out to support)
* [ ] [Security review done](https://developer.matomo.org/guides/security-in-piwik#checklist)
* [ ] [Wording review done](https://developer.matomo.org/guides/pull-request-reviews#translations-wording-review-done)
* [ ] [Code review done](https://developer.matomo.org/guides/pull-request-reviews#code-review-done)
* [ ] [Tests were added if useful/possible](https://developer.matomo.org/guides/pull-request-reviews#tests-were-added-if-usefulpossible)
* [ ] [Reviewed for breaking changes](https://developer.matomo.org/guides/pull-request-reviews#reviewed-for-breaking-changes)
* [ ] [Developer changelog updated if needed](https://developer.matomo.org/guides/pull-request-reviews#developer-changelog-updated-if-needed)
* [ ] [Documentation added if needed](https://developer.matomo.org/guides/pull-request-reviews#documentation-added-if-needed)
* [ ] Existing documentation updated if needed

### CODE REVIEW

File: ClientHints.php [hunk 1]
```diff
@@ -188,10 +188,12 @@ public function getOperatingSystemVersion(): string
     public function getBrandList(): array
     {
         if (\count($this->fullVersionList)) {
-            return \array_combine(
-                \array_column($this->fullVersionList, 'brand'),
-                \array_column($this->fullVersionList, 'version')
-            ) ?: [];
+            $brands = \array_column($this->fullVersionList, 'brand');
```
Comment (sanchezzzhak) [inline ClientHints.php:191]: = $this->fullVersionList['brand'] ?? []

File: ClientHints.php [hunk 2]
```diff
@@ -188,10 +188,12 @@ public function getOperatingSystemVersion(): string
     public function getBrandList(): array
     {
         if (\count($this->fullVersionList)) {
```
Comment (sanchezzzhak) [inline ClientHints.php:190]: add the is_array check before using count

File: ClientHints.php (hunk 1 above)
Comment (sgiehl) [reply ClientHints.php:191]: That isn't the same. It extracts the `brand` column form all arrays contained in `$this->fullVersionList`

File: ClientHints.php (hunk 2 above)
Comment (sgiehl) [reply ClientHints.php:190]: will do that.

Comment (sanchezzzhak) [reply ClientHints.php]: maybe `count() > 0` is superfluous, I don't know exactly what will happen if 2 arrays are empty in `array_combine`

**Review by sanchezzzhak — APPROVED** [review]

**Pull request merged**